ruby/lib/clone_handler.rb

```diff
@@ -0,0 +1,122 @@
+require 'json'
+require 'net/http'
+
+class CloneHandler
+  PER_PAGE = 30.freeze
+  BASE_URL = 'https://api.github.com'
+
+  def self.get_username
+    Kernel.puts 'Please enter your Github username'
+    STDOUT.flush
+    STDIN.gets.chomp
+  end
+
+  def initialize(username)
+    @username = username
+    @errors = []
+  end
+
+  def clone
+    validate_clonable
+    if errors?
+      print_errors
+      return false
+    end
+
+    Kernel.system(final_command)
+  end
+
+  private
+
+  attr_reader :username, :errors
+
+  def final_command
+    @final_command ||= clone_commands.flatten.join(' & ')
+  end
+
+  def clone_commands
+    repo_info.map { |info| build_repo_clone_command(info) }
+  end
+
+  def repo_info
+    @repo_info ||= page_count.times.map { |index| repo_info_for_page(index + 1) }.flatten.compact
```

Comment:
Note that this collects all repo info across pages. This all occurs before clone commands are built.